tests/mock_http_runtime/mock/mod.rs

```diff
@@ -1,33 +1,58 @@
-use dyn_clone::DynClone;
 use pocket_ic::common::rest::{CanisterHttpRequest, CanisterHttpResponse};
-use std::{collections::VecDeque, fmt::Debug};
+use std::fmt::Debug;
 
 pub mod json;
 
-#[derive(Clone, Debug, Default)]
-pub struct MockHttpOutcalls(VecDeque<MockHttpOutcall>);
+#[derive(Debug, Default)]
+pub struct MockHttpOutcalls(Vec<MockHttpOutcall>);
 
 impl MockHttpOutcalls {
     pub fn push(&mut self, mock: MockHttpOutcall) {
-        self.0.push_back(mock);
+        self.0.push(mock);
     }
-}
 
-impl Iterator for MockHttpOutcalls {
-    type Item = MockHttpOutcall;
+    pub fn pop_matching(&mut self, request: &CanisterHttpRequest) -> Option<MockHttpOutcall> {
+        let matching_positions = self
+            .0
+            .iter()
+            .enumerate()
+            .filter_map(|(i, mock)| {
+                if mock.request.matches(request) {
+                    Some(i)
+                } else {
+                    None
+                }
+            })
+            .collect::<Vec<_>>();
+
+        match matching_positions.len() {
+            0 => None,
+            1 => Some(self.0.remove(matching_positions[0])),
```

Comment:
nit: since we don't care about the order of mocks we can use `swap_remove`, which is O(1) instead of O(n).
```suggestion
            1 => Some(self.0.swap_remove(matching_positions[0])),
```